src/Tools/dotnet-monitor/DiagnosticsMonitorCommandHandler.cs

```diff
@@ -98,19 +165,130 @@ public IHostBuilder CreateHostBuilder(IConsole console, string[] urls, string[]
                 })
                 .ConfigureWebHostDefaults(webBuilder =>
                 {
-                    webBuilder.UseUrls(urls);
-                    webBuilder.UseStartup<Startup>();
+                    webBuilder.ConfigureKestrel((context, options) =>
+                    {
+                        //Note our priorities for hosting urls don't match the default behavior.
+                        //Default Kestrel behavior priority
+                        //1) ConfigureKestrel settings
+                        //2) Command line arguments (--urls)
+                        //3) Environment variables (ASPNETCORE_URLS, then DOTNETCORE_URLS)
+
+                        //Our precedence
+                        //1) Environment variables (ASPNETCORE_URLS, DotnetMonitor_Metrics__Endpoints)
+                        //2) Command line arguments (these have defaults) --urls, --metricUrls
+                        //3) ConfigureKestrel is used for fine control of the server, but honors the first two configurations.
+
+                        string hostingUrl = context.Configuration.GetValue<string>(WebHostDefaults.ServerUrlsKey);
+                        if (!string.IsNullOrEmpty(hostingUrl))
+                        {
+                            urls = ConfigurationHelper.SplitValue(hostingUrl);
+                        }
+
+                        var metricsOptions = new MetricsOptions();
+                        context.Configuration.Bind(MetricsOptions.ConfigurationKey, metricsOptions);
+
+                        if (metricsOptions.Enabled)
+                        {
+                            metricUrls = ProcessMetricUrls(metricUrls, metricsOptions);
+                            urls = urls.Concat(metricUrls).ToArray();
+                        }
+
+                        bool boundListeningPort = false;
+
+                        //Workaround for lack of default certificate. See https://github.com/dotnet/aspnetcore/issues/28120
+                        options.Configure(context.Configuration.GetSection("Kestrel")).Load();
+
+                        //By default, we bind to https for sensitive data (such as dumps and traces) and bind http for
+                        //non-sensitive data such as metrics. We may be missing a certificate for https binding. We want to continue with the
+                        //http binding in that scenario.
```

Comment:
Do we allow binding to HTTP if we have authentication turned on?  Would it be possible to not allow HTTP binding at all, expecially when authentication is enabled?

Reply:
We would probably just need to pipe the `noAuth` to `ProcessMetricUrls` and just evaluate:

``` c#
if (!noAuth || (metricsOptions.Providers.Count > 0) &&
                (!metricsOptions.AllowInsecureChannelForCustomMetrics) &&
                (metricsOptions.Providers.Any(provider =>
                    !Monitoring.EventPipe.MonitoringSourceConfiguration.DefaultMetricProviders.Contains(provider.ProviderName, StringComparer.OrdinalIgnoreCase))))
{
    // make metricUrls https.
}
```

Reply:
Per the security review, we are allowing auth + http. We do want to add additional warnings in that scenario.

Reply:
I'm a little nervous about allowing auth over an HTTP connection without lots of warnings and alarm bells.  Could we disallow it that configuration until we have said warnings?

CC @shirhatti

Reply:
I have added the warning.